admin-dev/themes/new-theme/js/pages/order/view/order-product-renderer.js

```diff
@@ -195,6 +196,29 @@ export default class OrderProductRenderer {
     this.togglePaginationControls();
   }
 
+  updateNumPerPage(numPerPage) {
+    const $rows = $(OrderViewPageMap.productsTable).find('tr[id^="orderProduct_"]');
+    const $tablePagination = $(OrderViewPageMap.productsTablePagination);
+    const numPages = Math.ceil($rows.length / numPerPage);
```

Comment:
And if you have 0 rows ? or if numPerPage = 0 ?

Reply:
`numPerPage` is bounded on the controller side, and I didn't think about empty products But you're right I should add some safeguard in the js code as well

Reply:
Done!